src/Events/Admin/Onboarding/Controller.php

```diff
@@ -88,6 +88,53 @@ public function handle_onboarding_page_dismiss(): void {
 		$this->container->make( Landing_Page::class )->handle_onboarding_page_dismiss();
 	}
 
+	/**
+	 * Redirects users to the Guided Setup page when accessing any TEC settings or management page for the first time.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function redirect_tec_pages_to_guided_setup(): void {
+		// Get some information about the page that we are trying to access.
+		$page      = tribe_get_request_var( 'page' );
+		$post_type = tribe_get_request_var( 'post_type' );
+
+		// Do not redirect if they have older versions and are probably already set up.
+		$tec_versions = (array) tribe_get_option( 'previous_ecp_versions', [] );
+		if ( count( $tec_versions ) > 1 ) {
+			return;
+		}
+
+		// Do not redirect if the target is not The Events Calendar-related admin pages.
+		if ( ! str_contains( $post_type, 'tribe' ) ) {
```

Comment:
I would become more specific to be honest.

i would do something like:

$post_types = apply_filters( 'tec_events_....', [ 'tribe_events', 'tribe_another' ..... ] );

if ( ! in_array() ) {
 return;
}